### PR TITLE
nbdime: init at 1.0.6

### PR DESCRIPTION
**Pull request by tbenst**
###### Motivation for this change

Adds a useful tool for diffing and merging of Jupyter notebooks. Also updated `jsonschema`--had to remove tests as couldn't get them to run due to new test framework based on `tox`. Perhaps @domenkozar has an idea?

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Also not sure if I did this recompilation test correctly...
```
~/c/nixpkgs> nix-shell -p nix-review --run "nix-review wip"
these paths will be fetched (0.08 MiB download, 0.38 MiB unpacked):
  /nix/store/w7xgdpblmq59blg2qzcdj951rgl2h6v0-bash-interactive-4.4-p23-dev
  /nix/store/x9vlpm4wyzzjwjdmcm3sbpbq2lvs8n0r-nix-review-2.0.0
copying path '/nix/store/w7xgdpblmq59blg2qzcdj951rgl2h6v0-bash-interactive-4.4-p23-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/x9vlpm4wyzzjwjdmcm3sbpbq2lvs8n0r-nix-review-2.0.0' from 'https://cache.nixos.org'...
$ git fetch --force https://github.com/NixOS/nixpkgs master:refs/nix-review/0
From https://github.com/NixOS/nixpkgs
 * [new branch]              master     -> refs/nix-review/0
$ git worktree add /home/tyler/.cache/nix-review/rev-5fd27a66239c96ef088c45aa1a7fb40b74cf897e-dirty/nixpkgs efcdac63fee4280cdc276362199503b6acb75df2
Preparing worktree (detached HEAD efcdac63fee)
Checking out files: 100% (18930/18930), done.
HEAD is now at efcdac63fee nixos/pantheon: add geoclue application configuration
$ nix-env -f /home/tyler/.cache/nix-review/rev-5fd27a66239c96ef088c45aa1a7fb40b74cf897e-dirty/nixpkgs -qaP --xml --out-path --show-trace
No diff detected, stopping review...
$ git worktree prune
```

### CODE REVIEW

**Comment by teto** [comment]
better use nix run nixpkgs.nix-review. As for the tox testing, nixpkgs' hypothesis seems to remove the tox.ini . IIRC, you can look into the tox.ini and run the test directly in the checkPhase.

**Comment by tbenst** [comment]
@teto thanks! Tests aren't all passing so made an issue here: https://github.com/Julian/jsonschema/issues/562

**Review by teto — CHANGES_REQUESTED** [review]

**Comment by teto** [comment]
Until an upstream fix, you could run only specific tests with `-k` https://docs.python.org/3.7/library/unittest.html#command-line-interface, there are some examples in nixpkgs

**Comment by tbenst** [comment]
@teto thx! incorporated all suggestions.

**Comment by tbenst** [comment]
All tests now passing for `jsonschema`

**Comment by tbenst** [comment]
Unfortunately, a number of python packages hardcode `jsonshema~=2.6`. This can be fixed by adding `sed -i "s/jsonschema~=2.6/jsonschema/" requirements/base.txt` to `postPatch`. Should I do all of this in this pull request or make a new one for jsonschema?

**Comment by teto** [comment]
> Unfortunately, a number of python packages hardcode

How many ? If many we could bump jsonschema while keeping the version of 2.6 at the same time. I am not fond of patching the requirement since it may induce some other problems.

**Comment by tbenst** [comment]
@teto ~17 packages rely on `jsonschema`, although I'm not sure how many specify a version. I know that `aws-sam-translator` and `cfn-lint` rely on 2.6, but haven't checked any others. I tried having two versions, but get the following error when trying to build:

```
Found duplicated packages in closure for dependency 'jsonschema': 
  jsonschema 3.0.1 (/nix/store/di3bkdy954ngkpwnnxi2cqwdwvqh76rj-python3.7-jsonschema-3.0.1/lib/python3.7/site-packages)
  jsonschema 2.6.0 (/nix/store/5z3pax2qlzqyy4xbx0wn0yc3gdv1kpil-python3.7-jsonschema-2.6.0/lib/python3.7/site-packages)

Package duplicates found in closure, see above. Usually this happens if two packages depend on different version of the same dependency.
```

**Comment by teto** [comment]
Ok not sure what to advise then, @FRidh should know better.

**Comment by tbenst** [comment]
I was able to get work locally by using `buildPythonApplication` in nbdime and  creating this overlay. Could do something similar for this pull request? That way can have two versions of jsonschema as @teto suggested.

```
(self: super: {
  nbdimeOverrides = python-self: python-super: {
    jsonschema = python-super.callPackage /Computer/packages/jsonschema.nix { };
      };
    pythonNbdime = super.python37.override {packageOverrides = self.nbdimeOverrides;};
  nbdime = super.callPackage /Computer/packages/nbdime.nix { python3Packages = self.pythonNbdime.pkgs; };
  })

```

**Comment by tbenst** [comment]
@teto turns out the poetry pkg had to deal with jsonschema issue too. They resolved by making local nix derivation. I `callPackage` on this unlisted derivation. Let me know what you think..?

**Review by jonringer — APPROVED** [review]

File: pkgs/development/python-modules/nbdime/default.nix [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+, GitPython
+, notebook
+, jinja2
+}:
+
+buildPythonPackage rec {
+  pname = "nbdime";
+  version = "1.0.6";
+  disabled = !isPy3k;
+  jsonschema = callPackage ../poetry/jsonschema.nix;
```
[the remark this answers is not in the document]
Comment (tbenst) [reply pkgs/development/python-modules/nbdime/default.nix]: @jonringer this package requires jsonschema 3, but many other packages require 2. nixpkgs currently only has jsonschema 2. I think this is a similar case to poetry as most people will use nbdime for the command line diff commands on Jupyter notebooks. Originally this pull request bumped jsonschema but it seemed to break lots of other pacakages

[the remark this answers is not in the document]
Comment (jonringer) [reply pkgs/development/python-modules/nbdime/default.nix]: master has jsonschmea 3

```
$ nix-build -A python3Packages.jsonschema
/nix/store/1zd7cnkb7lng5532hi1zlqj8dmm7n8lv-python3.7-jsonschema-3.1.1
```

[the remark this answers is not in the document]
Comment (jonringer) [reply pkgs/development/python-modules/nbdime/default.nix]: should be able to just import it like any other package

**Comment by jonringer** [comment]
@GrahamcOfBorg build python37Packages.nbdime python38Packages.nbdime

**Review by jonringer — APPROVED** [review]
`nix-review` passes on NixOS
diff LGTM
executables seem to work

```
[4 built, 17 copied (86.7 MiB), 12.2 MiB DL]
https://github.com/NixOS/nixpkgs/pull/61606
2 package were build:
python37Packages.nbdime python38Packages.nbdime
```

darwin build is broken by ipython, unrelated to this PR

**Pull request merged**